### PR TITLE
Issue 738 sitemap xml

### LINKED ISSUE #738: Add sitemap.xml for search engine indexing
Background

The Engineering Guidance and Standards site currently provides a human-readable sitemap page, but there does not appear to be an XML sitemap available for search engines and external crawlers.

Problem

- Search engines typically use sitemap.xml to discover and index site content.
- The site contains a growing number of principles, standards, patterns and supporting pages.
- Without a generated sitemap.xml there is an increased risk that content is not discovered or indexed efficiently.
- There is currently no obvious sitemap.xml generation process documented within the repository.

Proposal

Implement automated generation of sitemap.xml as part of the Eleventy build process and publish it at the site root.

Suggested scope

- Generate sitemap.xml during site build.
- Include all published principles, standards, patterns and supporting content pages.
- Exclude draft, archived or redirected content where appropriate.
- Document the sitemap generation approach in the repository README or technical documentation.
- Add a test/check to ensure sitemap.xml is produced during build.

Acceptance Criteria

- A sitemap.xml file is published at:
  https://engineering.homeoffice.gov.uk/sitemap.xml
- All published content pages are included.
- sitemap.xml is automatically regenerated during each deployment.
- The generation approach is documented for contributors.

### PR DESCRIPTION
**Pull request by nattrassHO**
This pull request adds automated generation of a `sitemap.xml` for the site, improves sitemap URL handling for different environments, and introduces tests to ensure the sitemap is correctly built and XML is properly escaped. It also sets the correct `SITE_ROOT` environment variable for production builds and refactors some test utilities for clarity.

**Sitemap generation and configuration:**

* Adds a new `docs/sitemap.xml.njk` template to generate `sitemap.xml` from published Markdown pages, including last-modified dates and excluding generated exports.
* Updates the Eleventy configuration and documentation to use the `SITE_ROOT` environment variable for sitemap URLs, defaulting to `http://localhost:8080/` for local builds and setting it to the production URL in the deployment workflow. (`.eleventy.js`, `.github/workflows/deploy-to-core-cloud.yml`, `README.md`) [[1]](diffhunk://#diff-c306e0a99961a16f5c5c83996caa0958b94006d97f97475049ea3a08036bb5b0L27-R27) [[2]](diffhunk://#diff-f61dcfafa5332a0d91400908005ea443600a9e7fc6a307ed6dd48580f504f6ccR20) [[3]](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R56-R61)

**Testing and validation:**

* Adds Playwright end-to-end tests to verify that `sitemap.xml` is generated, contains the correct content pages, last-modified dates, and excludes exports.
* Refactors test utility for building test root URLs in `tests/e2e/exports.spec.js` for clarity and reuse.

**XML escaping:**

* Implements a new `escapeXml` filter to safely escape special XML characters in URLs and text for the sitemap.
* Adds unit tests to ensure the `escapeXml` filter works correctly with various input types and edge cases.

# Code change
I can confirm:
## Accessibility considerations

- [ X] This change will not change layouts, page structures or anything else that might impact accessibility

## Commit signing

- [ X] All commits are signed with a key that can be verified by GitHub. [GitHub guidance on signing commits](https://docs.github.com/en/enterprise-cloud@latest/authentication/managing-commit-signature-verification/signing-commits)

### CODE REVIEW

**Comment by nattrassHO** [comment]
My assumption is that there is a SITE_ROOT variable available to the github pipeline for both staging and prod when you are reviewing this code.

File: README.md [hunk 1]
```diff
@@ -53,6 +53,14 @@ http-server -p 8080
...
 Now you can preview the site on http://localhost:8080
 
+### Sitemap XML
+
+The Eleventy build generates `_site/sitemap.xml` at the site root. The sitemap is built from the published Markdown pages in the Eleventy sitemap collection, which includes content pages such as principles, standards, patterns and supporting pages. Entries include a last-modified date when the page defines `date` metadata. Generated exports are excluded from the sitemap.
+
+Sitemap URLs use the `SITE_ROOT` environment variable. Local builds default to `http://localhost:8080/`.
+
+The production and staging deployments read `SITE_ROOT` from their `core-cloud` and `core-cloud-staging` GitHub environments. Configure each environment variable with its public site URL before deploying.
+
```
Comment (jeff-horton-ho-sas) [inline README.md]: This should be documented in `technical-docs/architecture-decision-records/technical-decision-log.md`

File: .eleventy.js [hunk 2]
```diff
@@ -24,7 +25,7 @@ function injectGitSha(eleventyConfig, gitHubRepositoryUrl) {
 }
 
 export default async function(eleventyConfig) {
-    const _siteRoot = process.env.SITE_ROOT ?? 'http://localhost:8080/';
+    const _siteRoot = normalizeSiteRoot(process.env.SITE_ROOT || 'http://localhost:8080/');
```
Comment (jeff-horton-ho-sas) [inline .eleventy.js]: What issue is this fixing. It feels like it's adding a lot of boilerplate for a non-issue.

`??` was correct here - if the SITE_ROOT has been explicitly set to the empty string or some other falsey value - that should be respected.

IMO putting this in it's own file is less clear than a line like:
```
const _siteRoot = (process.env.SITE_ROOT ?? 'http://localhost:8080/').replace(/\/*$/, '/');
```

Should also rename this to  siteRoot (`_siteRoot` implies it's not used)

File: .github/workflows/deploy-to-core-cloud.yml [hunk 3]
```diff
@@ -23,6 +23,8 @@ jobs:
     name: Build, test, and sync to S3
     environment:
       name: core-cloud
+    env:
+      SITE_ROOT: ${{ vars.SITE_ROOT }}
```
Comment (jeff-horton-ho-sas) [inline .github/workflows/deploy-to-core-cloud.yml]: No need to set this for the whole job
```suggestion
```

File: .github/workflows/deploy-to-core-cloud.yml [hunk 4]
```diff
@@ -45,6 +47,12 @@ jobs:
...
         run: npm run test:unit
 
+      - name: Validate site root
+        if: ${{ env.SITE_ROOT == '' }}
+        run: |
+          echo "SITE_ROOT must be configured for the core-cloud environment"
+          exit 1
+
       - name: Build site
         run: npm run build
```
Comment (jeff-horton-ho-sas) [inline .github/workflows/deploy-to-core-cloud.yml]:
```suggestion
      - name: Validate site root
        if: ${{ vars.SITE_ROOT == '' }}
        run: |
          echo "SITE_ROOT environment variable must be configured for the core-cloud environment"
          exit 1

      - name: Build site
        run: npm run build
        env:
          SITE_ROOT: ${{ vars.SITE_ROOT }}
```

File: .github/workflows/deploy-to-staging-core-cloud.yml [hunk 5]
```diff
@@ -23,6 +23,8 @@ jobs:
     name: Build, test, and sync to S3
     environment:
       name: core-cloud-staging
+    env:
+      SITE_ROOT: ${{ vars.SITE_ROOT }}
```
Comment (jeff-horton-ho-sas) [inline .github/workflows/deploy-to-staging-core-cloud.yml]:
```suggestion
```

File: .github/workflows/deploy-to-staging-core-cloud.yml [hunk 6]
```diff
@@ -45,6 +47,12 @@ jobs:
...
         run: npm run test:unit
 
+      - name: Validate site root
+        if: ${{ env.SITE_ROOT == '' }}
+        run: |
+          echo "SITE_ROOT must be configured for the core-cloud-staging environment"
+          exit 1
+
       - name: Build site
         run: npm run build
```
Comment (jeff-horton-ho-sas) [inline .github/workflows/deploy-to-staging-core-cloud.yml]:
```suggestion
      - name: Validate site root
        if: ${{ vars.SITE_ROOT == '' }}
        run: |
          echo "SITE_ROOT environment variable must be configured for the core-cloud-staging environment"
          exit 1

      - name: Build site
        run: npm run build
        env:
          SITE_ROOT: ${{ vars.SITE_ROOT }}
```
